docs/rules/no-unnecessary-act.md

```diff
@@ -0,0 +1,117 @@
+# Disallow wrapping Testing Library utils or empty callbacks in `act` (`testing-library/no-unnecessary-act`)
+
+> ⚠️ The `act` method is only available on the following Testing Library packages:
+>
+> - `@testing-library/react` (supported by this plugin)
+> - `@testing-library/preact` (not supported yet by this plugin)
+> - `@testing-library/svelte` (not supported yet by this plugin)
+
+## Rule Details
+
+This rule aims to avoid the usage of `act` to wrap Testing Library utils just to silence "not wrapped in act(...)" warnings.
+
+All Testing Library utils are already wrapped in `act`. Most of the time, if you're seeing an `act` warning, it's not just something to be silenced, but it's actually telling you that something unexpected is happening in your test.
+
+Additionally, wrapping empty callbacks in `act` is also an incorrect way of silencing "not wrapped in act(...)" warnings.
+
+Code violations reported by this rule will pinpoint those unnecessary `act`, helping to understand when `act` actually is necessary.
+
+Example of **incorrect** code for this rule:
+
+```js
+// ❌ wrapping things related to Testing Library in `act` is incorrect
+import {
+  act,
+  render,
+  screen,
+  waitFor,
+  fireEvent,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+// ...
+
+act(() => {
+  render(<Example />);
+});
+
+await act(async () => waitFor(() => {}));
+
+act(() => screen.getByRole('button'));
+
+act(() => {
+  fireEvent.click(element);
+});
+
+act(() => {
+  userEvent.click(element);
+});
+```
+
+```js
+// ❌ wrapping empty callbacks in `act` is incorrect
+import { act } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+// ...
+
+act(() => {});
+
+await act(async () => {});
+```
+
+```js
+// ❌ wrapping things related to Testing Library in React DOM Test Utils `act` is also incorrect
```

Comment:
Isn't this the same as the code examples above?

Reply:
Almost. The only difference is that `act` is imported from `react-dom/test-utils` rather than `@testing-library/react`. I guess I can add a comment with an arrow pointing that difference or make it clear changing the example.

Reply:
Couldn't you remove this duplicate stuff and add a comment above indicating it's the same for both.

Reply:
I've removed the duplicated block and added a simple comment about `'react-dom/test-utils'`. Thanks!